GoogleForCreators/web-stories-wp
# Fix Color Picker: Saving text background color saves text color instead

## Context



Currently, when you click a text element's background color and try to save the background color to the local or global palettes, the element's text color is saved instead. 

## Summary


I am making use of the `changedStyle` prop in the `ColorPicker` component which is currently used for implementing changes in the element's attributes, to detect from which attribute the color picker was triggered from and save the correct color / background color accordingly. 

## Relevant Technical Choices



## To-do



## User-facing changes



| Before | After |
|--------|-------|
|   ![144714656-96795a83-f5e4-44a5-81fe-2f772cd04dec](https://user-images.githubusercontent.com/276316/146688881-db026420-eb58-4fc4-afef-124f65cae386.gif) |  ![Screen Recording 2021-12-19 at 22 33 05](https://user-images.githubusercontent.com/276316/146688775-4005a3e6-b952-474b-8551-262aea42a632.gif)  |


## Testing Instructions






This PR can be tested by following these steps:

1. Insert text element
2. Change bg color
3. Try to save bg color
4. make sure the correct color is added to both the local and global colors.
5. Make sure saving the text color also works correctly.


## Reviews

### Does this PR have a security-related impact?


No

### Does this PR change what data or activity we track or use?


No

### Does this PR have a legal-related impact?


No

## Checklist




---



Fixes #9891
